packages/api/portfolios/application/getApplication.ts

```diff
@@ -0,0 +1,50 @@
+import "reflect-metadata";
+import { APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import middy from "@middy/core";
+import xssSanitizer from "../../utils/xssSanitizer";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import { ApiGatewayEventParsed } from "../../utils/eventHandlingTool";
+import { validateRequestShape } from "../../utils/shapeValidator";
+import { CORS_CONFIGURATION } from "../../utils/corsConfig";
+import { errorHandlingMiddleware } from "../../utils/errorHandlingMiddleware";
+import { Application, IApplicationCreate } from "../../../orm/entity/Application";
+import { createConnection } from "../../utils/database";
+import { ApplicationRepository } from "../../repository/ApplicationRepository";
+
+/**
+ * Retrieve an Application based on applicationId
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<Application>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  validateRequestShape<Application>(event);
+  // it seems that we don't even need the portfolioId for this operation
+  // we should bring this up to Jeff, and see what he thinks
+  const portfolioId = event.pathParameters?.portfolioId as string;
```

Comment:
Good point. This also happens for environment operations.

Reply:
The `portfolioId` is not technically required for the retrieval of the application since `applicationId` should be unique. However, the API design presents a logical and hierarchical set of endpoints that illustrates the relationships between the entities.  I don't think it would makes sense to change this for the simple GET operations simply because an implementation detail allows us to do so.

As Julius explained, we should check that the `id` combination is valid and we're not retrieving the requested application child if the wrong parent is specified in the path parameters.  In that case let's return a 400 status with generic _Invalid input_ result, while logging something more specific about the parent not containing the requested child or something similar.

Reply:
Making additional SELECTs to the database will only slow down the application as the database gets populated.

Reply:
We could accomplish the portfolioId validation locally without making another SELECT.